tests/unit/test_req_file.py

```diff
@@ -598,13 +598,14 @@ def make_var(name):
         with open(tmpdir.joinpath('req1.txt'), 'w') as fp:
             fp.write(template.format(*map(make_var, env_vars)))
 
+        session = PipSession()
```

Comment:
Changes in this file around `PipSession` seem unrelated?

Reply:
They're not -- PipSession now checks for an env var (specifically `PATHEXT` on Windows), and these were being constructed inside a monkey-patch of getenv (which was intended to test `parse_requirements`).

Reply:
Ah, I see. Can you add a comment explaining this, as it seems like a very subtle interaction that would be incredibly hard to diagnose if someone later decided to omit the extra variable as part of a tidy-up.

Actually, does that mean we're doing the subprocess call every time we construct a `PipSession`? Would it not be better to just calculate the rust version once, as it's not as if it's going to change?

Reply:
Done.

And yes, we do it once per PipSession. While we construct a few of them in tests, I think in `pip` itself, it's only ever constructed once, so I'm not sure it's a big deal?

Reply:
Yeah, probably not. Test suite times are dominated by running pip over and over, so a couple of extra rustc runs isn't going to harm anything.